transcom/mymove
# MB-7332 Prime can cancel shipment

## Description

This PR adds the `updateMTOShipmentStatus` endpoint to the Prime API so that the Prime can cancel shipments when the TOO requests it. 

## Reviewer's Notes

For some reason, when I merged with `master`, my linter decided to alphabetize all of the frontend imports. I think this is harmless, but if you want me to grab a frontend person to review them, let me know. 

## Setup

To test, first run your server:

```sh
make server_run
```

If you're testing with the Prime API Client, you'll also need to rebuild that binary:

```sh
rm -rf bin/prime-api-client && make bin/prime-api-client
```

Once you have your server running, you'll need to:

* Find some Prime-available shipments
* Mark at least one as "CANCELLATION_REQUESTED" using either the TOO interface, Support API, or a DB viewer
* Hit `fetchMTOUpdates` or `getMoveTaskOrder` to get the shipment's ID and eTag
* Hit `updateMTOShipmentStatus` with that ID, eTag, and the following body:

```json
{
  "status": "CANCELED"
}
```

Verify that you get a 200 response code.

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7332) for this change.
